### PR TITLE
Add How-to for CKF

### PR DESCRIPTION
**Pull request by XiaocongAi**
This PR edits/adds the How-to docs for truth fitting/CKF reconstruction.

### CODE REVIEW

**Comment by paulgessinger** [comment]
Would it be possible / make sense to add some plots of the root output files? To give people a better idea of what the expected output looks like?

**Comment by XiaocongAi** [comment]
That's indeed a good idea! Have to learn a little bit how to insert the plots though.

**Comment by paulgessinger** [comment]
You can put files next to the `.rst` files, or in `docs/figures/` and just commit them. Then you can reference them in the rST. You should have an eye on the file size not being overly large. I'd probably try to keep the filesize to about <200kb.

**Comment by XiaocongAi** [comment]
I have added some plots. :smiley:

File: docs/howto/run_truth_tracking.rst [hunk 1]
```diff
@@ -8,34 +8,113 @@ Acts must be build with activated examples and Pythia8 support
...
+
+The truth tracking will generate three root files (the name of those root files are currently not configurable via the command line) in the ``output-dir``:
+
+*   ``tracks.root``
+This includes a tree with one entry representing one trajectory. From this file, one could check the information of every measurement track state on the trajectory.
+
+*  ``performace_track_finder.root``
+This includes a tree showing performance of the truth track finding.
+
+*  ``peformance_track_fitter.root``
```
Comment (paulgessinger) [inline docs/howto/run_truth_tracking.rst:90]:
```suggestion
*  ``performance_track_fitter.root``
```

File: docs/howto/run_truth_tracking.rst [hunk 2]
```diff
@@ -8,34 +8,113 @@ Acts must be build with activated examples and Pythia8 support
...
+This includes a few histograms showing the residual&pull of the fitted perigee track parameters and efficiency plots showing the fitting efficiency etc.
```
Comment (paulgessinger) [inline docs/howto/run_truth_tracking.rst:91]:
```suggestion
This includes a few histograms showing the residual and pull of the fitted perigee track parameters and efficiency plots showing the fitting efficiency etc.
```

File: docs/howto/run_truth_tracking.rst (hunk 1 above)
Comment (XiaocongAi) [reply docs/howto/run_truth_tracking.rst:90]: Thank you! Changed.

File: docs/howto/run_truth_tracking.rst (hunk 2 above)
Comment (XiaocongAi) [reply docs/howto/run_truth_tracking.rst]: Thank you! Changed.

File: docs/howto/howto.rst [hunk 3]
```diff
@@ -6,3 +6,4 @@ How-to guides
 
    run_fatras
    run_truth_tracking
+   run_CKF_tracking
```
[the remark this answers is not in the document]
Comment (XiaocongAi) [reply docs/howto/howto.rst]: Ah, right. It's much better. Thank you!

File: docs/howto/run_CKF_tracking.rst [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+----------------------
+
+Run the CKF tracking tool that reads the simulation output (truth hits and truth particles), creates smeared
+measurements from the true hits, creates seeds (i.e. starting track parameters) from the pre-selected truth particles, 
+and run the CKF which will perform the track finding and track fitting simultaneously:
+
+Currently, there are two configurable criteria to select compatible source links on a surface with track parameters in CKF:
+
+* Global maximum chi2 of Kalman filtering. This could be set up via ``--ckf-slselection-chi2max``
+
```
[the remark this answers is not in the document]
Comment (XiaocongAi) [reply docs/howto/run_CKF_tracking.rst]: Sure!

**Pull request merged**